### PR TITLE
fix: findIndexFile return false if indexFiles option false

### PR DESCRIPTION
**Pull request by erictaylor**
`FastifyStaticOptions` `index` option is currently typed as `false | string[]`. When using `index: false` option you will encounter a `TypeError: indexFiles.find is not a function` error because the `findIndexFile` function assumes that the argument is always an array when the option can be `false`.

This adds a `.isArray` check before `.find()` usage.

Additionally, this updates the typing info for the `index` option to be `string[] | string | false` (adds `string`). This is to match actual usage and info on `index` usage in README.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added (n/a)
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by gajus — APPROVED** [review]

**Review by mcollina — CHANGES_REQUESTED** [review]
Can you please add a unit test?

**Comment by erictaylor** [comment]
@mcollina Added tests and updated `index` option typing to reflect `string` type as well.

File: index.d.ts [hunk 1]
```diff
@@ -88,7 +88,7 @@ export interface FastifyStaticOptions extends SendOptions {
   etag?: boolean;
   extensions?: string[];
   immutable?: boolean;
-  index?: string[] | false;
+  index?: string[] | string | false;
```
Comment (mcollina) [inline index.d.ts:91]: could you please update the tests for the types? We use tsd.

**Review by mcollina — CHANGES_REQUESTED** [review]
good work just one note!

File: index.d.ts (hunk 1 above)
Comment (erictaylor) [reply index.d.ts:91]: @mcollina does e914e64 work?

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by erictaylor** [comment]
@mcollina merge conflicts resolved.

**Pull request merged**